python/tchannel/tornado/peer.py

```diff
@@ -411,19 +432,39 @@ def __init__(self, peer, service, **kwargs):
         :param service:
             Name of the service being called through this peer. Defaults to
             an empty string.
+        :param arg_scheme
+            arg scheme type
+        :param retry
+            retry type
+        :param parent_tracing
+            tracing span from parent request
         """
         assert peer, "peer must not be None"
         service = service or ''
 
         self.peer = peer
         self.service = service
-        self.parent_tracing = kwargs.get('parent_tracing', None)
+        self.parent_tracing = parent_tracing
+        self.headers = {
+            'as': arg_scheme or ArgSchemeType.DEFAULT,
+            're': retry or RetryType.DEFAULT,
+        }
+
+        # keep all arguments for retry purpose.
+        # not retry if hostport is set.
+        self._hostport = hostport
+        self._score_threshold = score_threshold
         # service name is not stored in peer because the same peer may be
         # used to call multiple services if it's being used for request
         # forwarding
 
     @gen.coroutine
-    def send(self, arg1, arg2, arg3, traceflag=False, headers=None):
+    def send(self, arg1, arg2, arg3,
+             traceflag=False,
+             headers=None,
+             retry_times=Request.MAX_RETRY_TIMES,
+             timeout_per_req=Request.TIMEOUT,
```

Comment:
Just call this `timeout_millis` or `timeout_secs` depending on what the unit of time is.